app/lib/activitypub/activity/create.rb

```diff
@@ -114,16 +115,18 @@ def process_status_params
         sensitive: @object['sensitive'] || false,
         visibility: visibility_from_audience,
         thread: replied_to_status,
-        conversation: conversation_from_uri(@object['conversation']),
+        conversation: conversation_from_context,
```

Comment:
I'm pretty worried about re-purposing `Conversation` and changing the way it's federated, with no regards with backward compatibility. Isn't that going to break threads (and thus the “mute conversation” feature) when going back and forth between an old (well, current) Mastodon version and one with that patch?

Also, what about changing the visibility of a public thread to a non-public one down the road? Especially if the thread has not been started with this patch.

Reply:
My main objective with the code so far (in the draft stage) was to prove (mainly to myself) that the functionality was possible to implement.